Move pvutil.go from e2e package to framework package

msau42 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://github.com/kubernetes/kubernetes/blob/master/CONTRIBUTING.md and developer guide https://github.com/kubernetes/kubernetes/blob/master/docs/devel/development.md
2. If you want *faster* PR reviews, read how: https://github.com/kubernetes/kubernetes/blob/master/docs/devel/faster_reviews.md
3. Follow the instructions for writing a release note: https://github.com/kubernetes/kubernetes/blob/master/docs/devel/pull-requests.md#release-notes
-->

**What this PR does / why we need it**:  

This PR moves pvutil.go to the e2e/framework package.

I am working on a PV upgrade test, and would like to use some of the wrapper functions in pvutil.go.  However, the upgrade test is in the upgrade package, and not the e2e package, and it cannot import the e2e package because it would create a circular dependency.  So pvutil.go needs to be moved out of e2e in order to break the circular dependency.  This is a refactoring name change, no logic has been modified.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: 

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access) 
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`. 
-->
NONE

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/41373)
<!-- Reviewable:end -->

jsafrane commented on test/e2e/common/pvutil.go:
```diff
@@ -593,3 +594,13 @@ func createClientPod(c clientset.Interface, ns string, pvc *v1.PersistentVolumeC
 	return clientPod
 }
+
+func GetGCECloud() (*gcecloud.GCECloud, error) {
```
Is there a better place for this than `pvutil.go`?

msau42 replied:
Maybe framework/util.go would be a better place? It seems to already have a few gce methods in it.

Actually, now I'm thinking pvutil.go should go under framework/ instead of common/.  It looks like there are other _util.go files for other components there.

msau42 committed:
Move pvutil.go into framework/ folder.

This is the first step in a two step change.  First is to move the file.
 The second step is to change the package and callers.  It needs to be
 two steps in order to correctly transfer commit history.

msau42 committed:
Move pv_util.go into framework package so that it can be used in tests
not in the e2e package.

msau42 commented:
Moved pvutil.go to framework package since that's where other component's utility functions are located in.

jsafrane commented:
/lgtm

msau42 commented:
cc @spiffxp.  Would you be able to take a look at this for approval?

spiffxp commented:
@msau42 unfortunately I can't approve, not listed as an approver... however, one comment

This isn't really brought about by your PR, but I just want to point out this code looks like it's mostly volume-related.  To me, that would imply that @kubernetes/sig-storage-pr-reviews would be the right set of people to be listed as approvers or reviewers.

What about making this live in a sibling package to framework, with an OWNERS file that calls out some sig-storage members, rather than stuffing inside and falling through to test/e2e OWNERS?  /cc @kubernetes/sig-testing-pr-reviews

msau42 commented:
Ah sorry, I misread the OWNERS file.

I could make it its own package, but I saw other _util.go files for other components in framework, so thought this would be more consistent.

spiffxp commented:
yeah I mean, it's probably scope creep for this PR, but thought I would just mention it if anyone else has an opinion

lgtm otherwise

fejta commented:
@apelisse @grodrigues3 any idea why I'm not getting assigned if the system wants me to approve this?
@spiffxp sent you https://github.com/kubernetes/kubernetes/pull/41920 to give you approval for this dir

fejta commented:
/approve

apelisse commented:
@fejta https://github.com/kubernetes/test-infra/issues/1687 is not fixed yet. We're not sure how and if we want to get there yet.
